examples/intro_causal_graphs.py

```diff
@@ -0,0 +1,235 @@
+    "y",
+    gcm.AdditiveNoiseModel(
+        prediction_model=MyCustomModel(1),
+        noise_model=gcm.ScipyDistribution(stats.binom, p=0.8, n=1),
+    ),
+)
+causal_model.set_causal_mechanism(
+    "w",
+    gcm.AdditiveNoiseModel(
+        prediction_model=MyCustomModel(1),
+        noise_model=gcm.ScipyDistribution(stats.binom, p=0.5, n=1),
+    ),
+)
+causal_model.set_causal_mechanism(
+    "x",
+    gcm.AdditiveNoiseModel(
+        prediction_model=MyCustomModel(1),
+        noise_model=gcm.ScipyDistribution(stats.binom, p=0.5, n=1),
+    ),
+)
+
+# Fit here would not really fit parameters, since we don't do anything in the fit method.
+# Here, we only need this to ensure that each FCM has the correct local hash (i.e., we
+# get an inconsistency error if we would modify the graph afterwards without updating
+# the FCMs). Having an empty data set is a small workaround, since all models are
+# pre-defined.
+gcm.fit(causal_model, pd.DataFrame(columns=["x", "y", "z", "w", "xy"]))
+
+# sample the observational data
+data = gcm.draw_samples(causal_model, num_samples=500)
+
+print(data.head())
+print(pd.Series({col: data[col].unique() for col in data}))
+
+# note the graph shows colliders, which is a collision of arrows
+# for example between ``x`` and ``z`` at ``y``.
+draw(G)
+
+# %%
+# Causal Directed Ayclic Graphs (DAG): Also known as Causal Bayesian Networks
+# ---------------------------------------------------------------------------
+#
+# Causal DAGs represent Markovian SCMs, also known as the "causally sufficient"
+# assumption, where there are no unobserved confounders in the graph.
+print(G)
+
+# One can query the parents of 'y' for example
+print(list(G.predecessors("y")))
+
+# Or the children of 'x'
+print(list(G.successors("xy")))
+
+# Using the graph, we can explore d-separation statements, which by the Markov
+# condition, imply conditional independences.
+# For example, 'z' is d-separated from 'x' because of the collider at 'y'
+print(f"'z' is d-separated from 'x': {nx.d_separated(G, {'z'}, {'x'}, set())}")
+
+# Conditioning on the collider, opens up the path
+print(f"'z' is d-separated from 'x' given 'y': {nx.d_separated(G, {'z'}, {'x'}, {'y'})}")
+
+# %%
+# Acyclic Directed Mixed Graphs (ADMG)
+# ------------------------------------
+#
+# ADMGs represent Semi-Markovian SCMs, where there are possibly unobserved confounders
+# in the graph. These unobserved confounders are graphically depicted with a bidirected edge.
+
+# We can construct an ADMG from the DAG by just setting 'xy' as a latent confounder
+admg = pywhy_graphs.set_nodes_as_latent_confounders(G, ["xy"])
+
+# Now there is a bidirected edge between 'x' and 'y'
+draw(admg)
+
+# Now if one queries the parents of 'y', it will not show 'xy' anymore
+print(list(admg.predecessors("y")))
+
+# The bidirected edges also form a cluster in what is known as "confounded-components", or
+# c-components for short.
+print(f"The ADMG has c-components: {list(admg.c_components())}")
```

Comment:
Is "c_components" a well-known expression in the science community or would this benefit from the verbose name "confounded_components".

Reply:
I think the string can be more explicit, but in general anyone working in causality from graphical model perspective would get exposed to c-components (imo), but open to changing the property name too.